encoding/ssz/merkleize.go

```diff
@@ -196,3 +197,42 @@ func ConstructProof(hasher Hasher, count, limit uint64, leaf func(i uint64) []by
 
 	return
 }
+
+// MerkelizeVector uses our optimized routine to hash a list of 32-byte
+// elements.
+func MerkelizeVector(elements [][32]byte, length uint64) [32]byte {
+	depth := Depth(length)
+	// Return zerohash at depth
+	if len(elements) == 0 {
+		zerohash := trie.ZeroHashes[depth]
+		return zerohash
+	}
+	for i := 0; i < int(depth); i++ {
+		layerLen := len(elements)
+		oddNodeLength := layerLen%2 == 1
+		if oddNodeLength {
+			zerohash := trie.ZeroHashes[i]
+			elements = append(elements, zerohash)
+		}
+		outputLen := len(elements) / 2
+		htr.VectorizedSha256(elements, elements[:outputLen])
```

Comment:
There's no need to pass here `elements[:outputLen]` we can pass elements and it will be overwritten up to the right length, no need to create a new slice header.